File: backend/internal/testfixture/work_code_test.go
package testfixture

import "testing"

func TestWorkCodeUsesReservedPrefixRanges(t *testing.T) {
	tests := []struct {
		prefix WorkCodePrefix
		want   string
	}{
		{prefix: PrefixRJ, want: "RJ00000007"},
		{prefix: PrefixBJ, want: "BJ00000007"},
		{prefix: PrefixVJ, want: "VJ00000007"},
		{prefix: PrefixCC, want: "CC00000007"},
	}
	for _, test := range tests {
		if got := WorkCode(test.prefix, 7); got != test.want {
			t.Fatalf("WorkCode(%q, 7) = %q, want %q", test.prefix, got, test.want)
		}
	}
}

func TestWorkCodeAtSpansReservedPrefixRanges(t *testing.T) {
	tests := map[int]string{
		0:   "RJ00000000",
		99:  "RJ00000099",
		100: "BJ00000000",
		399: "CC00000099",
	}
	for index, want := range tests {
		if got := WorkCodeAt(index); got != want {
			t.Fatalf("WorkCodeAt(%d) = %q, want %q", index, got, want)
		}
	}
}

func TestWorkCodeRejectsValuesOutsideReservedRanges(t *testing.T) {
	for _, call := range []func(){
		func() { WorkCode("ZZ", 0) },
		func() { WorkCode(PrefixRJ, -1) },
		func() { WorkCode(PrefixRJ, 100) },
		func() { WorkCodeAt(400) },
	} {
		func() {
			defer func() {
				if recover() == nil {
					t.Fatal("synthetic work-code constructor did not panic")
				}
			}()
			call()
		}()
	}
}

File: frontend/src/test-support/workCode.test.ts
import { describe, expect, it } from "vitest";

import { syntheticWorkCode, syntheticWorkCodeAt } from "./workCode";

describe("synthetic work-code fixtures", () => {
  it.each([
    ["RJ", "RJ00000007"],
    ["BJ", "BJ00000007"],
    ["VJ", "VJ00000007"],
    ["CC", "CC00000007"],
  ] as const)("constructs the reserved %s range", (prefix, expected) => {
    expect(syntheticWorkCode(prefix, 7)).toBe(expected);
  });

  it("spans reserved prefix ranges for high-cardinality fixtures", () => {
    expect(syntheticWorkCodeAt(0)).toBe("RJ00000000");
    expect(syntheticWorkCodeAt(99)).toBe("RJ00000099");
    expect(syntheticWorkCodeAt(100)).toBe("BJ00000000");
    expect(syntheticWorkCodeAt(399)).toBe("CC00000099");
  });

  it("rejects values outside the reserved ranges", () => {
    expect(() => syntheticWorkCode("RJ", -1)).toThrow(RangeError);
    expect(() => syntheticWorkCode("RJ", 100)).toThrow(RangeError);
    expect(() => syntheticWorkCodeAt(400)).toThrow(RangeError);
  });
});
